tsl/src/remote/connection.c

```diff
@@ -2348,12 +2348,8 @@ remote_connection_end_copy(TSConnection *conn, TSConnectionError *err)
 		ExecStatusType status = PQresultStatus(res);
 		if (status != PGRES_COMMAND_OK)
 		{
-			success =
-				fill_result_error(err,
-								  ERRCODE_CONNECTION_EXCEPTION,
-								  psprintf("invalid result status '%s' when ending remote COPY",
-										   PQresStatus(status)),
-								  res);
+			success = false;
+			remote_connection_get_result_error(res, err);
```

Comment:
In other functions (like in `remote_connection_begin_copy`) we use `fill_result_error` to report connection errors. Using this function, we can add additional information about the error location that might be useful for future error investigation.

```suggestion
                fill_result_error(err,
                                                  ERRCODE_CONNECTION_FAILURE,
                                                  "Invalid status when ending remote COPY",
                                                  res);
```

Reply:
I found that fill_result_error overwrites the original error, i.e. it doesn't report what the error on the connection was. Let's use this one instead, we also use it in other similar places such as row-by-row data fetcher.

Reply:
I was mentioning this, because `remote_connection_get_result_error` simply calls `fill_result_error` ([link](https://github.com/timescale/timescaledb/blob/533a138ae22ddabf8a576d23dcb8f88a842fb6e5/tsl/src/remote/connection.c#L881)) with an empty `errmsg`.

Reply:
Yeah... It's weird that passing a non-empty message overwrites the remote message, can't tell how this can happen from the code. But I've definitely seen it show "Invalid status PGRES_FATAL_ERROR" instead of the remote error text, which is bad.